api/health/worker.go

```diff
@@ -204,10 +265,41 @@ func (w *worker) runCheck(ctx context.Context, wg *sync.WaitGroup, name string,
 		}
 
 		if prevResult.Error == nil {
```

Comment:
generally this function has become a bit hard to follow with those ifs/elses

Reply:
definitely open to suggestions... I couldn't find a cleaner way to do it

Reply:
not sure if this is simpler, but IMO at least complexity reduced down to a helper function lol:
https://github.com/ava-labs/avalanchego/pull/1596/files